packages/bun-plugin-svelte/src/plugin.ts

```diff
@@ -0,0 +1,72 @@
+import { basename } from "node:path";
+import type { BuildConfig, Transpiler } from "bun";
+import { getBaseCompileOptions, getBaseModuleCompileOptions, hash, type SvelteOptions } from "./options";
+import { compile, type CompileOptions, type CompileResult, type ModuleCompileOptions } from "svelte/compiler";
+
+const virtualNamespace = "bun-svelte";
+
+export class SveltePluginService {
+  private compileOptions: CompileOptions;
+  private compileModuleOptions: ModuleCompileOptions;
+
+  private ts: Transpiler;
+  /** Virtual CSS modules created by Svelte components. Keys are import specifiers. */
+  private css: Map<string, VirtualCSSModule>;
+
+  constructor(options: SvelteOptions, config: Partial<BuildConfig>) {
+    this.compileOptions = getBaseCompileOptions(options, config);
+    this.compileModuleOptions = getBaseModuleCompileOptions(options, config);
+
+    this.ts = new Bun.Transpiler({ loader: "ts" });
+    this.css = new Map();
+  }
+
+  public compileComponent(source: string, filename: string, hmr: boolean, side?: "client" | "server"): CompileResult {
+    const generate = this.compileOptions.generate ?? side;
+    const result = compile(source, {
+      ...this.compileOptions,
+      generate,
+      filename,
+      hmr,
+    });
+
+    var { js, css } = result;
+    if (css?.code && generate != "server") {
+      const uid = `${basename(filename)}-${hash(filename)}-style`.replaceAll(`"`, `'`);
+      const virtualName = virtualNamespace + ":" + uid + ".css";
+      this.css.set(virtualName, { sourcePath: filename, source: css.code });
+      js.code += `\nimport "${virtualName}";`;
+    }
+
+    return result;
+  }
+
+  public async compileModule(source: string, filename: string, side?: "client" | "server"): Promise<CompileResult> {
+    const generate = this.compileModuleOptions.generate ?? side;
+    if (filename.endsWith(".ts")) {
```

Comment:
Probably not important, but what about `.mts` and `.cts` extensions?

Reply:
Fixed to `.endsWith("ts")`